sdk/ml/azure-ai-ml/tests/dsl/e2etests/test_dsl_pipeline.py

```diff
@@ -1480,6 +1478,11 @@ def pipeline_distribution_components(job_in_number, job_in_path):
         created_job: PipelineJob = client.jobs.create_or_update(
             pipeline, experiment_name=experiment_name, continue_on_step_failure=True
         )
+        # Theoretically, we should keep the invalid name in request body,
+        # as component name valid in azureml-components maybe invalid in azure-ai-ml.
+        # So we leave this validation to server-side for now.
+        assert mpi_func._to_rest_object().properties.component_spec["name"] == invalid_component_name
```

Comment:
@wangchao1230 please help confirm this. Another solution is to automatically correct the invalid name on client-side and maybe give a warning

Reply:
can we raise exception for v2 components in client side when name is invalid?

Reply:
Of course. The problem is (what we tested here), if a component is anonymous, previously we won't raise exception for an invalid name as the name will be overwritten. So raising exception directly may lead to potential regression.

Reply:
Let's keep current behavior.